### PR TITLE
Adds Abridged accounts implementation in handler

### PR DESCRIPTION
**Pull request by ChrisJBurns**
- adds Abridged accounts implementation in AccountsHandler
- updates the references of the new abridged method in the controller class
- adds and improves the unit test coverage of the new scenarios
- improves javaDocs where needed

Resolves: SFA-567

### CODE REVIEW

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java [hunk 1]
```diff
@@ -32,17 +36,27 @@ public DocumentInfoResponse getAccountsData(String resourceLink) throws HandlerE
...
-        // TODO: The accountsType variable shall be used in the implementation of abridged logic
         AccountType accountsType = getAccountType(accounts);
 
+        String abridgedAccountLink = getAccountLink(accounts, accountsType);
+        try {
+            // TODO: abridgedAccountData will be used as part of the implementation of converting
+            // TODO: the data object to a json string
+            AbridgedAccountsApi abridgedAccountData = accountsService.getAbridgedAccounts(abridgedAccountLink);
+        } catch (ServiceException e) {
+            throw new HandlerException(e.getMessage(), e.getCause());
```
Comment (jsimons1989) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java]: Can you log a message here with the available details of the account that was affected by this error (account ID, account type, etc) please?

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java [hunk 2]
```diff
@@ -32,17 +36,27 @@ public DocumentInfoResponse getAccountsData(String resourceLink) throws HandlerE
             throw new HandlerException(e.getMessage(), e.getCause());
         }
 
-        // TODO: The accountsType variable shall be used in the implementation of abridged logic
         AccountType accountsType = getAccountType(accounts);
 
+        String abridgedAccountLink = getAccountLink(accounts, accountsType);
```
Comment (jsimons1989) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java]: The account type is being retrieved from the links to then retrieve the link using the account type. Is the account type going to be used for anything else or can this be simplified to just get the non-self link?

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java (hunk 1 above)
Comment (ChrisJBurns) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java]: Refined logging will be done in a separate PR as explain in the PR description as well as the refined exception handling. I have submitted basic logging so this should be fine, but remember that the logging isn't going to be complete in these PRs until there is a PR that addressing logging entirely.

Comment (jsimons1989) [reply document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java]: Logging is a non-functional requirement that should exist in each PR. I agree that we should review/improve it over time as we use the logs and it won't be perfect the first time but it needs to be included.

**Pull request merged**